pkg/automation/automation.go

```
@@ -298,128 +302,12 @@ func AirgapInstall(appSlug string, additionalFiles map[string][]byte) error {
-		cleanup(&licenseSecret, verifiedLicense.Spec.AppSlug)
-		return errors.Wrapf(err, "failed to check if license already exists for app %s", verifiedLicense.Spec.AppSlug)
-	}
-	if existingLicense != nil {
-		resolved, err := kotslicense.ResolveExistingLicense(verifiedLicense)
-		if err != nil {
-			logger.Error(errors.Wrap(err, "failed to resolve existing license conflict"))
-		}
-		if !resolved {
-			cleanup(&licenseSecret, verifiedLicense.Spec.AppSlug)
-			return errors.Errorf("license already exists for app %s", verifiedLicense.Spec.AppSlug)
-		}
-	}
-
-	instParams, err := kotsutil.GetInstallationParams(kotsadmtypes.KotsadmConfigMap)
-	if err != nil {
-		return errors.Wrap(err, "failed to get existing kotsadm config map")
-	}
-
-	desiredAppName := strings.Replace(verifiedLicense.Spec.AppSlug, "-", " ", 0)
-	upstreamURI := fmt.Sprintf("replicated://%s", verifiedLicense.Spec.AppSlug)
-
-	a, err := store.GetStore().CreateApp(desiredAppName, upstreamURI, string(license), verifiedLicense.Spec.IsAirgapSupported, instParams.SkipImagePush, instParams.RegistryIsReadOnly)
-	if err != nil {
-		return errors.Wrap(err, "failed to create app record")
-	}
-
-	// airgap data is the airgap manifest + app specs + image list laoded from configmaps
-	airgapData, err := getAirgapData(clientset, verifiedLicense)
-	if err != nil {
-		return errors.Wrapf(err, "failed to load airgap data for %s", verifiedLicense.Spec.AppSlug)
-	}
 
-	if len(airgapData) == 0 {
-		return errors.Errorf("failed to find airgap automation data")
-	}
-
-	for k, v := range additionalFiles {
-		airgapData[k] = v
-	}
-
-	// Images have been pushed and there is airgap app data available, so this is an airgap install.
-	airgapFilesDir, err := ioutil.TempDir("", "headless-airgap")
+	appSlug, err = installLicenseSecret(clientset, licenseSecret, additionalFiles)
 	if err != nil {
-		return errors.Wrap(err, "failed to create temp dir")
+		logger.Error(errors.Wrapf(err, "failed to install license for app %s", appSlug))
```

why not return an error here instead of logging it?